mitmproxy/tools/web/app.py

```diff
@@ -654,6 +656,43 @@ def get(self):
         self.write(State.get_json(self.master))
 
 
+class ProcessList(RequestHandler):
+    @staticmethod
+    def process_to_dict(process):
+        return {
+            "is_visible": process.is_visible,
+            "executable": process.executable,
+            "is_system": process.is_system,
+            "display_name": process.display_name,
+        }
+
+    @staticmethod
+    def get_json():
+        processes = mitmproxy_rs.active_executables()
+        processes_json = [ProcessList.process_to_dict(process) for process in processes]
+        return {"processes": processes_json}
+
+    def get(self):
+        self.write(ProcessList.get_json())
+
+
+class ProcessImage(RequestHandler):
+    def get(self):
+        path = self.get_query_argument("path", None)
+
+        if not path:
+            self.set_status(400)
+            self.write({"error": "Missing 'path' parameter."})
+            return
+
+        try:
+            icon_bytes = mitmproxy_rs.executable_icon(pathlib.Path(path))
```

Comment:
Passing `path` directly does not work?
(I'm mostly asking because the mitmproxy_rs API docs indicate it should, and if it doesn't we should fix the docs)

Reply:
it works! You're right!